viewing_party/party.py

```diff
@@ -1,23 +1,142 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if not (title and genre and rating):
+        return None
+    else:
+        movie_dict["title"]= title
+        movie_dict["genre"]= genre
+        movie_dict["rating"]= rating
+        return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+    
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data  
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+            return user_data
+    return user_data
+        
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    sum = 0.0
+    if not user_data["watched"]:
+        return sum
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    average = sum / len(user_data["watched"])
+    return average
+
+def get_most_watched_genre(user_data):
+    genre_frequency_dict = {}
+    if not user_data["watched"]:
+        return None
+    for movie in user_data["watched"]:
+        if not movie["genre"] in genre_frequency_dict.keys():
+            genre_frequency_dict[movie["genre"]]= 1
+        else:
+            genre_frequency_dict[movie["genre"]] += 1
+    
+    most_watched_genre = max(genre_frequency_dict, key = genre_frequency_dict.get)
+    return most_watched_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    user_watched_list = []
+    friend_watched_list = []
+    for movie in user_data["watched"]:
+        user_watched_list.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            friend_watched_list.append(friend_movie["title"])
+
+    user_set = set(user_watched_list)
+    friend_set = set(friend_watched_list)
+    unique_set = user_set - friend_set
+
+    unique_list = []
+    for title in unique_set:
+        for movie in user_data["watched"]:
+            if title == movie["title"]:
+                unique_list.append(movie)
+    return unique_list

```

Comment:
I like the way you leveraged sets to find the unique movies on line 74. Great job trying something new! The one thing to think about here is that the sets you are making deal only with the title of each movie (As we know, sets cannot hold mutable items). While there is nothing inherently wrong with this, it forces us to go back through the user's "watched" list in a second nested loop on lines 77-80, which is just an extra step we don't necessarily need. A different approach (which could be used in other functions as well) would be to make a copy of the user_data["watched"] list. Then as you go through the friends lists, you could simply remove any duplicates from the user_data["watched"] copy. In the end, your copy should be holding the correct subset of movies!